File: Legacy/Objective-C/Neocom/Neocom/UIImage+NC.m
//
//  UIImage+NC.m
//  Neocom
//
//  Created by Artem Shimanski on 20.11.16.
//  Copyright © 2016 Artem Shimanski. All rights reserved.
//

#import "UIImage+NC.h"
#import "UIColor+CS.h"

@implementation UIImage (NC)

+ (instancetype) imageWithColor:(UIColor*) color {
	UIGraphicsBeginImageContextWithOptions(CGSizeMake(1, 1), NO, 1);
	[color setFill];
	[[UIBezierPath bezierPathWithRect:CGRectMake(0, 0, 1, 1)] fill];
	UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
	UIGraphicsEndImageContext();
	return [image resizableImageWithCapInsets:UIEdgeInsetsZero resizingMode:UIImageResizingModeTile];
}

+ (instancetype) searchFieldBackgroundImageWithColor:(UIColor*) color {
	UIGraphicsBeginImageContextWithOptions(CGSizeMake(28, 28), NO, [[UIScreen mainScreen] scale]);
	[color setFill];
	UIBezierPath* path = [UIBezierPath bezierPathWithRoundedRect:CGRectMake(0, 0, 28, 28) cornerRadius:5];
	[path fill];
	UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
	UIGraphicsEndImageContext();
	return [image resizableImageWithCapInsets:UIEdgeInsetsMake(5, 5, 5, 5) resizingMode:UIImageResizingModeStretch];
}

@end

File: Legacy/Neocom/Neocom/UIImage+NC.swift
//
//  UIImage+NC.swift
//  Neocom
//
//  Created by Artem Shimanski on 05.12.16.
//  Copyright © 2016 Artem Shimanski. All rights reserved.
//

import UIKit

extension UIImage {
	class func image(color: UIColor, size: CGSize = CGSize(width: 1, height: 1), scale: CGFloat = 1) -> UIImage {
		UIGraphicsBeginImageContextWithOptions(size, false, scale);
		color.setFill()
		UIBezierPath(rect: CGRect(origin: .zero, size: size)).fill()
		let image = UIGraphicsGetImageFromCurrentImageContext()!;
		UIGraphicsEndImageContext();
		return image.resizableImage(withCapInsets: UIEdgeInsets.zero, resizingMode: UIImageResizingMode.tile)
	}
	
	class func searchFieldBackgroundImage(color: UIColor) -> UIImage {
		UIGraphicsBeginImageContextWithOptions(CGSize(width: 28, height: 28), false, UIScreen.main.scale);
		color.setFill()
		UIBezierPath(roundedRect: CGRect(x: 0, y: 0, width: 28, height: 28), cornerRadius: 5).fill()
		let image = UIGraphicsGetImageFromCurrentImageContext()!;
		UIGraphicsEndImageContext();
		return image.resizableImage(withCapInsets: UIEdgeInsets(top: 5, left: 5, bottom: 5, right: 5), resizingMode: UIImageResizingMode.stretch)
	}
}
